src/sql/workbench/contrib/notebook/test/browser/outputProcessor.test.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import * as op from 'sql/workbench/contrib/notebook/browser/models/outputProcessor';
+import { JSONObject } from 'sql/workbench/services/notebook/common/jsonext';
+import { nbformat as nbformat } from 'sql/workbench/services/notebook/common/nbformat';
+import { getRandomString } from 'vs/editor/test/common/model/linesTextBuffer/textBufferAutoTestUtils';
+
+
+suite('OutputProcessor functions', function (): void {
+	suite('getData', async function (): Promise<void> {
+		for (const outputType of ['execute_result', 'display_data', 'update_display_data', 'stream', 'error'] as nbformat.OutputType[]) {
+			const text = getRandomString(1, 10);
+			const output: nb.ICellOutput = {
+				output_type: outputType
+			};
+			if (nbformat.isError(output)) {
```

Comment:
Instead of having one huge block with tons of nested ifs it would be great if you could just simply split this up into multiple tests. For example I don't see why the error and stream have to be a part of this loop - they don't share any code paths.

Reply:
will do.

Reply:
split up into several blocks. One for reach types of tests. Thanks for the suggestion.